SPHINXsys/src/shared/bodies/body_relation.cpp

```diff
@@ -117,16 +117,16 @@ namespace SPH
 	ContactBodyRelation::ContactBodyRelation(SPHBody* sph_body, RealBodyVector contact_sph_bodies)
 		: BaseContactBodyRelation(sph_body, contact_sph_bodies)
 	{
-		initializaiton();
+		initialization();
 	}
 	//=================================================================================================//
 	ContactBodyRelation::ContactBodyRelation(SPHBody* sph_body, BodyPartVector contact_bodyparts)
 		: BaseContactBodyRelation(sph_body, contact_bodyparts)
 	{
-		initializaiton();
+		initialization();
 	}
 	//=================================================================================================//
-	void ContactBodyRelation::initializaiton()
+	void ContactBodyRelation::initialization()
 	{
 		for (size_t k = 0; k != contact_bodies_.size(); ++k) 
 		{
```

Comment:
get_search_ranges_.push_back(new SearchRangeMultiResolution
and 
get_contact_neighbors_.push_back(new NeighborRelationContact
have no reource cleaning. considering using unique_ptr and get_search_ranges_.emplace_back to avoid this kind of neglection

Reply:
These are one shot memory location. I will later check where I can use smart pointer type in one specialized branch.